### ISSUE TITLE
15. Формирование экзаменационных билетов по дисциплине.

### ISSUE THREAD

**Issue by ZinovNikita** [nzt94/Tema15#1]
Основные задачи:
1. Создать приложение на Java, используя IDE Eclipce.
2. Приложение должно соответствовать требованиям:
2.1 Формирование вопросов, билетов и дисциплин
2.2 Возможность редактирования, удаления вопросов, билетов, дисциплин
2.3 Возможность генерации файла rtf для распечатки.
3. К приложению приложить пояснительную записку, в которой описать все выполненные шаги:
3.1 Объединение команды из 4 человек во главе с "менеджером".
3.2 Распределить роли участников в команде
3.3 Организовать групповую разработку с помощью технологии Git (github.com)
3.4 В соответствии с распределенными ролями, реализовать приложении.

**Comment by FedorovskayaMM** [comment]
Может использовать базу данных, которая будет состоять из 3 таблиц Билеты, Вопросы и Дисциплины?

**Comment by ZinovNikita** [comment]
Да, интересная мысль, только я боюсь, что БД реализованы не кроссплатформенно, поэтому предлагаю реализовать парсинг csv файлов, которые можно редактировать в Excel. а связь между файлами также, как и в БД, через id.

**Comment by FedorovskayaMM** [comment]
Хорошо, но было бы лучше сделать базу данных...

**Comment by ZinovNikita** [comment]
Да, но предлагаю сделать пока так, а если найдем исходники для баз данных, то реализуем базу, только в ТЗ сказано, что приложение, должно запускаться на "любой" системе. Нужно будет это учесть.

**Comment by ZinovNikita** [comment]
Я там создал класс таблиц, есть предложения

**Comment by FedorovskayaMM** [comment]
может, добавить возможность добавления и удаления строк итп.

**Comment by ZinovNikita** [comment]
а, ну да, точно... Я просто еще не придумал как использовать класс этот.
Думаю сделать еще класс Баз данных, в котором будут все таблицы. А еще нужно реализовать так чтобы все работы со строками были в массивах, а не в файлах, а посл изменения массива, перезаписывать файл. Что скажешь?

**Comment by FedorovskayaMM** [comment]
Да, я вечером посмотрю, попробую github установить и подправить что-нибудь)))

**Comment by FedorovskayaMM** [comment]
Создала заглушки на форме, для работы с таблицами. Жду обработчик таблиц...

**Comment by ZinovNikita** [comment]
Почти закончили, создал экспорт rtf, теперь нужно форму сделать. может потом еще что-либо сделать

**Comment by FedorovskayaMM** [comment]
Ладно, я вечером форму сделаю

**Comment by FedorovskayaMM** [comment]
Я решила отказаться от поля "Тип вопроса"

**Comment by ZinovNikita** [comment]
Ладно, я тем более реализовал формирование шаблона без оглядки на это поле

**Comment by FedorovskayaMM** [comment]
Возникли проблемы при выборке по id. объект choice имеет только свой идентификатор (0-n) и название, а id может идти не по порядку после удаления записи. Нужно как-то решить эту проблему.

**Comment by ZinovNikita** [comment]
Исправил я скрытый массив добавил который связывает id строки и индекс в choice

**Comment by ZinovNikita** [comment]
Нужно сделать еще удаление данных на форме и возможность использования своего шаблона, который выбирается в диалоге. нужно сделать а форме как-то так, чтобы пока не выбран шаблон, нельзя экспортировать файл или при экспорте спросить шаблон, если он не задан

**Comment by FedorovskayaMM** [comment]
Посмотрю вечером

**Comment by FedorovskayaMM** [comment]
нужно как-то информировать как создавать шаблоны

**Comment by ZinovNikita** [comment]
я решил весь функционал разместить равномерно во вкладках верхнего меню
